src/net.h

```diff
@@ -1408,6 +1408,11 @@ class CConnman
     uint16_t GetDefaultPort(Network net) const;
     uint16_t GetDefaultPort(const std::string& addr) const;
 
+    /**
+     * Add entries to `m_unused_i2p_sessions` if needed.
+     */
```

Comment:
The `AddI2PSessionsIfNeeded` function is declared with `EXCLUSIVE_LOCKS_REQUIRED(!m_unused_i2p_sessions_mutex)`, which indicates that the function should not be called while holding the `m_unused_i2p_sessions_mutex`. However, the function is called within `CConnman::OpenNetworkConnection`, which is marked with the same lock requirement, potentially leading to a deadlock if the lock is held elsewhere in the call stack.